thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java

```diff
@@ -239,7 +242,18 @@ public Response evaluate(
     // see TE-1172 - ensure the dto passed to the authManager is complete, not only a reference {id:1234}
     final AlertDTO alertDTO = optional(alert.getId()).map(this::get).orElseGet(() -> toDto(alert));
     authorizationManager.ensureCanEvaluate(principal, alertDTO);
-    return Response.ok(alertEvaluator.evaluate(request)).build();
+
+    final AlertEvaluationApi results = alertEvaluator.evaluate(request);
+    final HashMap<String, DetectionEvaluationApi> allowedEvals = new HashMap<>();
+    results.getDetectionEvaluations().forEach((k,v) -> {
+      final EnumerationItemDTO item = ApiBeanMapper.toEnumerationItemDTO(v.getEnumerationItem());
+      if (authorizationManager.hasAccess(principal, item, AccessType.READ)) {
+        allowedEvals.put(k, v);
+      }
+    });
```

Comment:
Filtering after the evaluation gives the expected output in the UI. I only see enumeration items that I have permission to see. 

I'd prefer to filter before evaluating, so we aren't throwing away work.